tests/ballerina-test/src/test/java/org/ballerinalang/test/net/http/HttpCachingClientTest.java

```diff
@@ -0,0 +1,441 @@
+        // Not testing for null arguments since in code, this function is always called after a null check
+        BValue[] returns = BRunUtil.invoke(compileResult, "isAStrongValidator", new BValue[]{new BString("abc")});
+        Assert.assertTrue(((BBoolean) returns[0]).booleanValue());
+
+        returns = BRunUtil.invoke(compileResult, "isAStrongValidator", new BValue[]{new BString("12345")});
+        Assert.assertTrue(((BBoolean) returns[0]).booleanValue());
+
+        returns = BRunUtil.invoke(compileResult, "isAStrongValidator", new BValue[]{new BString("W/abc")});
+        Assert.assertFalse(((BBoolean) returns[0]).booleanValue());
+
+        returns = BRunUtil.invoke(compileResult, "isAStrongValidator", new BValue[]{new BString("w/abc")});
+        Assert.assertTrue(((BBoolean) returns[0]).booleanValue());
+
+        returns = BRunUtil.invoke(compileResult, "isAStrongValidator", new BValue[]{new BString("")});
+        Assert.assertTrue(((BBoolean) returns[0]).booleanValue());
+    }
+
+    @Test
+    public void testHasAWeakValidator() {
+        final String lastModifiedHeader = "Thu, 01 Mar 2018 15:36:34 GMT";
+        final String etagHeader = "1sps79e:q0efehi8";
+
+        BStruct validationResponse = BCompileUtil.createAndGetStruct(compileResult.getProgFile(),
+                                                                     PROTOCOL_PACKAGE_HTTP,
+                                                                     IN_RESPONSE);
+        HTTPCarbonMessage validationResponseMsg = HttpUtil.createHttpCarbonMessage(false);
+        validationResponseMsg.setProperty(HTTP_STATUS_CODE, 200);
+        validationResponseMsg.setHeader(LAST_MODIFIED, lastModifiedHeader);
+        validationResponseMsg.setHeader(ETAG, etagHeader);
+        initInboundResponse(validationResponse, validationResponseMsg);
+
+        // Not testing with a null validation response since this function is not called without a null check
+
+        BValue[] inputArg = {validationResponse, new BString(etagHeader)};
+        BValue[] returns = BRunUtil.invoke(compileResult, "hasAWeakValidator", inputArg);
+        Assert.assertTrue(((BBoolean) returns[0]).booleanValue());
+
+        inputArg[1] = new BString(null);
+        returns = BRunUtil.invoke(compileResult, "hasAWeakValidator", inputArg);
+        Assert.assertTrue(((BBoolean) returns[0]).booleanValue());
+
+        validationResponseMsg.removeHeader(LAST_MODIFIED);
+        initInboundResponse(validationResponse, validationResponseMsg);
+        inputArg[1] = new BString(etagHeader);
+        returns = BRunUtil.invoke(compileResult, "hasAWeakValidator", inputArg);
+        Assert.assertFalse(((BBoolean) returns[0]).booleanValue());
+
+        inputArg[1] = new BString("W/" + etagHeader); // passing a weak validator
+        returns = BRunUtil.invoke(compileResult, "hasAWeakValidator", inputArg);
+        Assert.assertTrue(((BBoolean) returns[0]).booleanValue());
+    }
+
+    @Test
```

Comment:
May be we can provide descriptions for these test annotations why these tests were added. Eg:- @Test(description = "This test following functionality etc")